erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -197,6 +197,23 @@ def validate_sales_order(self):
 				if chk:
 					throw(_("Maintenance Schedule {0} exists against {1}").format(chk[0][0], d.sales_order))
 
+	def validate_items_table_change(self):
+		doc_before_save = self.get_doc_before_save()
+		if not doc_before_save:
+			return
+		for prev_item, item in zip(doc_before_save.items, self.items):
+			fields = ['item_code', 'start_date', 'end_date', 'periodicity', 'sales_person', 'no_of_visits', 'serial_no']
+			schedule_generated = False
+			for field in fields:
+				b_doc = prev_item.as_dict()
+				doc = item.as_dict()
+				if cstr(b_doc[field]) != cstr(doc[field]):
+					schedule_generated = True
+					self.generate_schedule()
+					break
+			if schedule_generated:
+				break
```

Comment:
any reason to track this flag? Why not just return from the whole function call? 

Also `validate_items_table_change` shouldn't be generating schedule, the name is misleading. Either it should return bool indicating change in items, or name should be changed 😅